DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/task_sequences/README.md

```diff
@@ -55,70 +55,109 @@ You can also find more information about [troubleshooting build errors](/DirectP
+[[intel::use_stall_enable_clusters]] 
+void LoopC(int len) {
+  [[intel::initiation_interval(1)]]  
+  for (size_t i = 0; i < len; i++) {
+    int tmp = PipeBC::read();
+    tmp += i;
+    PipeCD::write(tmp);  
+  }
+}
 
-## Sample Structure
-The 2 different example designs in this sample perform similar operations.
+[[intel::use_stall_enable_clusters]] 
+void LoopD(int len) {
+  [[intel::initiation_interval(1)]]  
+  for (size_t i = 0; i < len; i++) {
+    int tmp0 = PipeCD::read();
+    int tmp1 = PipeAD::read();
+    int out = tmp0 + tmp1;
+    PipeOut::write(out);  
+  }
+}
 
-1. [Naive](naive/) This implementation uses arrays to pass data between loops and each loop are scheduled to run sequentially.
-2. [Task sequence](task_sequence/) This implementation uses inter-kernel pipes to pass data between loops and applies task sequences to schedule loops to run concurrently.
+///////////////////////////////////////
 
-## Build a Design
+struct OptimizedKernel {
+  int len;
 
+  void operator()() const {
+    sycl::ext::intel::experimental::task_sequence<LoopA> task_a;
+    sycl::ext::intel::experimental::task_sequence<LoopB> task_b;
+    sycl::ext::intel::experimental::task_sequence<LoopC> task_c;
+    sycl::ext::intel::experimental::task_sequence<LoopD> task_d;
 
-Use the appropriate TYPE parameter when running CMake to config which design to compile:
-| Example                                      | Directory             | Type (-DTYPE=) |
-|----------------------------------------------|-----------------------|----------------|
-| Naive                                        | naive/                | `NAIVE`        |
-| Task sequence                                | task-sequence/        | `TASK_SEQUENCE`|
+    task_a.async(len);
+    task_b.async(len);
+    task_c.async(len);
+    task_d.async(len);
+  }
+};
+```
+C++ functions `LoopA`, `LoopB`, `LoopC` and `LoopD` are the task functions with asynchronous activities defined. Each of the task functions is used to parameterize a specific task_sequence class to create four task_sequence object instances, `task_a`, `task_b`,`task_c` and `task_d`. Parallel invocation of the task functions is launched through `async()` method that accepts the same arguments as their respective task functions. The
+`async()` method call is non-blocking, and it returns before the asynchronous task functions invocation completes executing and potentially before the task functions even begins executing, as the return type from the `async()` method provides no implicit information on the execution status of task functions.
 
+You can see a concrete example of a kernel that naively uses four sequential loops in `naive/main.cpp`. This design is modified to use task sequences to run the loops concurrently in `task_sequence/main.cpp`.
```

Comment:
```suggestion
You can see a concrete example of a kernel that naively uses four sequential loops in `naive/main.cpp`. This design is modified to use task sequences to run the loops in parallel in `task_sequence/main.cpp`.
```

Reply:
@yuguen-intel I feel like we can go beyond saying 'concurrent' all over the place here and say 'in parallel'. Do you agree?

https://stackoverflow.com/questions/1050222/what-is-the-difference-between-concurrency-and-parallelism